base/tps/src/org/dogtagpki/server/tps/rest/ActivityService.java

```diff
@@ -164,44 +233,132 @@ protected void retrieveActivitiesWithoutVLV(
             Integer size,
             ActivityCollection response) throws Exception {
 
+        String method = "ActivityService.retrieveActivitiesWithoutVLV: ";
         // search without VLV
         Iterator<ActivityRecord> activities = database.findRecords(filter).iterator();
+        List<String> authorizedProfiles = getAuthorizedProfiles();
 
-        // TODO: sort results by date in reverse order
-
+        // count the total entries
         int i = 0;
+        for (; activities.hasNext(); i++)
+            activities.next();
```

Comment:
The `activities` is an `Iterator` and this loop will bring the iterator to the end of the list, so it will not be usable anymore. The original code was indeed a bit weird. Maybe we can use `List` so we don't have to deal with iterators anymore:
```
List<ActivityRecord> activities = (List<ActivityRecord>) database.findRecords(filter);
int total = activities.size();
```

Reply:
While it makes sense,  since I'm really not familiar with the area, I try to avoid making too much changes.  Could I just leave this alone for now to meet my deadline, and perhaps we could refactor later?  Same for some of the similar comments below.

Reply:
I don't think the current PR will work as is. You will need to restart the iterator after counting the total, so something like this:
```
Collection<ActivityRecord> activityCollection = database.findRecords(filter);

// count total
Iterator<ActivityRecord> activities = activityCollection.iterator();
int i = 0;
for (; activities.hasNext(); i++)
    activities.next();
int total = i;

// go to the start of the page
activities = activityCollection.iterator();
for (i=0; i < start && activities.hasNext(); i++)
    activities.next();
```
Then you'll need to return the page up to the provided size (instead of start + size):
```
for (i=0; i < size && activities.hasNext(); i++) {
    ActivityRecord record = activities.next();
    ...
}
```

Reply:
I just realized that the reason why it's been working for me was because it's the ActivityService.retrieveActivitiesWithVLV that gets called for both the web UI and the cli.
So how do you trigger ActivityService.retrieveActivitiesWithoutVLV ?  Since I didn't change the logic there, I suspect it was never called so the issue you raised was never discovered?

I'll need to know how to trigger this to test it if I were to make changes. Please let me know. Thanks!!

Reply:
I think you need to specify a search filter:
https://github.com/dogtagpki/pki/blob/3e3671241db609ab139df40301fde32dfe29d39c/base/tps/src/main/java/org/dogtagpki/server/tps/rest/ActivityService.java#L111-L115
Make sure the filter returns enough records so you can test the paging.

Reply:
@edewata my question was, how do you (either through the web UI, or the cli) get the system to call the ActivityService.retrieveActivitiesWithoutVLV without doing any code changes?  Which options(s) of either web ui or cli does that?

Reply:
Yes, there should be a `Search...` field at the top of the activities table in TPS UI.

For the CLI it's not documented on the wiki, but it's shown in the help message:
https://github.com/dogtagpki/pki/blob/master/base/tools/src/main/java/com/netscape/cmstools/logging/ActivityFindCLI.java#L48
so it can be used like this:
```
$ pki -n caadmin tps-activity-find <filter>
```
See also https://github.com/dogtagpki/pki/wiki/PKI-TPS-Activity-CLI

Reply:
@edewata  sorry I missed your response and just force pushed change that follows your original suggestion of using "List".
I thought I'd push it out while waiting for your response and didn't realize that you already did ;-).
I'll give it a test and let you know.  Thanks!